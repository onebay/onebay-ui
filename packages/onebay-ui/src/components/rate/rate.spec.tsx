import { mount } from '@vue/test-utils'
import Rate from './index'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(Rate, {
    slots,
    props: { ...values },
  })
}

describe('Rate Snap', () => {
  it('render initial Rate', () => {
    const wrapper = factory({ isTest: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Rate -- props className', () => {
    const wrapper = factory({ className: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Rate -- props customStyle', () => {
    const wrapper = factory({ customStyle: 'color:red;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Rate -- props size', () => {
    const wrapper = factory({ size: '10' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Rate -- props value', () => {
    const wrapper = factory({ value: 2 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Rate -- props max', () => {
    const wrapper = factory({ max: 10 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Rate -- props margin', () => {
    const wrapper = factory({ margin: 10 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Rate -- props float value', () => {
    const wrapper = factory({ value: 2.4 })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('Rate Event', () => {
  it('Rate onChange Event', () => {
    const onChange = jest.fn()
    const wrapper = factory({ value: 2, onChange: onChange })
    wrapper.find('.ob-rate__icon').trigger('click')
    expect(onChange).toBeCalled()
  })
})
