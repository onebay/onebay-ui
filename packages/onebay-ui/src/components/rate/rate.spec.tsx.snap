// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rate Snap render Rate -- props className 1`] = `
<div
  class="ob-rate test"
>
  
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  
</div>
`;

exports[`Rate Snap render Rate -- props style 1`] = `
<div
  class="ob-rate"
  style="color: red;"
>
  
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  
</div>
`;

exports[`Rate Snap render Rate -- props float value 1`] = `
<div
  class="ob-rate"
>
  
  <div
    class="ob-rate__icon ob-rate__icon--on"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--on"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--half"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  
</div>
`;

exports[`Rate Snap render Rate -- props margin 1`] = `
<div
  class="ob-rate"
>
  
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  
</div>
`;

exports[`Rate Snap render Rate -- props max 1`] = `
<div
  class="ob-rate"
>
  
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  
</div>
`;

exports[`Rate Snap render Rate -- props size 1`] = `
<div
  class="ob-rate"
>
  
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
      style="font-size: 10px;"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
        style="font-size: 10px;"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
      style="font-size: 10px;"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
        style="font-size: 10px;"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
      style="font-size: 10px;"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
        style="font-size: 10px;"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
      style="font-size: 10px;"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
        style="font-size: 10px;"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
      style="font-size: 10px;"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
        style="font-size: 10px;"
      />
    </div>
  </div>
  
</div>
`;

exports[`Rate Snap render Rate -- props value 1`] = `
<div
  class="ob-rate"
>
  
  <div
    class="ob-rate__icon ob-rate__icon--on"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--on"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  
</div>
`;

exports[`Rate Snap render initial Rate 1`] = `
<div
  class="ob-rate"
  istest="true"
>
  
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  <div
    class="ob-rate__icon ob-rate__icon--off"
  >
    <div
      class="ob-icon ob-icon-star-2"
    />
    <div
      class="ob-rate__left"
    >
      <div
        class="ob-icon ob-icon-star-2"
      />
    </div>
  </div>
  
</div>
`;
